ipa-core/src/protocol/prss/mod.rs

```diff
@@ -103,12 +61,25 @@ impl From<PrssIndex128> for u128 {
     }
 }
 
+impl TryFrom<u128> for PrssIndex128 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: u128) -> Result<Self, Self::Error> {
+        let value = u64::try_from(value)?;
+
+        Ok(Self {
+            index: u32::try_from(value >> 32).unwrap().into(),
+            offset: u32::try_from(value & u64::from(u32::MAX)).unwrap(),
```

Comment:
I have a new request...

Can we ensure that either the offset or the index is <= 2^11?  Probably the offset.  The analysis I did for PRSS indicates that we can't use this PRSS algorithm more than 2^43 times and this might be a good way to cap that.

Reply:
I can cap the `value` to be < 2^43, that seems easier than deciding which side of the index we want to cap. Also gives us the flexibility of giving more bytes to offset or index